src/endpoint/RestApi.js

```diff
@@ -0,0 +1,45 @@
+import HttpMethod from './HttpMethod.js';
+import env from '../env.js';
+
+const _baseURL = env.baseURL;
+
+
+const request = async (uri, method, body = undefined, timeout = 5000) => {
+  const config = { method };
+  const url = _baseURL + uri;
+
+  if (body) {
+    config.body = JSON.stringify(body);
+    config.headers = { 'Content-Type': 'application/json' };
+  }
+
+  let id = -1;
+  const race = await Promise.race([
+    new Promise(res => id = window.setTimeout(() => res(), timeout)),
+    fetch(url, config)
+  ]);
+
+  if (race instanceof Response) {
+    clearTimeout(id);
+    const contents = await race.json();
+
+    if (race.status === 500) new Error(contents.message);
+    if (race.status === 404) new Error('404');
+    return contents;
```

Comment:
```suggestion
    if (race.status === 500) throw new Error(contents.message);
    if (race.status === 404) throw new Error('404');
    return race.json();
```
코드 순서는 이렇게 해도 좋을 것 같아요!